lib/BC/Util.cpp

```diff
@@ -559,59 +566,61 @@ namespace {
 #define S(x) _S(x)
 #define MAJOR_MINOR S(LLVM_VERSION_MAJOR) "." S(LLVM_VERSION_MINOR)
 
-static const char *gSemanticsSearchPaths[] = {
-
-    // Derived from the build.
-    REMILL_BUILD_SEMANTICS_DIR_X86 "\0",
-    REMILL_BUILD_SEMANTICS_DIR_AARCH32 "\0",
-    REMILL_BUILD_SEMANTICS_DIR_AARCH64 "\0",
-    REMILL_BUILD_SEMANTICS_DIR_SPARC32 "\0",
-    REMILL_BUILD_SEMANTICS_DIR_SPARC64 "\0",
-    REMILL_INSTALL_SEMANTICS_DIR "\0",
+const std::vector< std::string > &DefaultSemanticsSearchPaths()
+{
+  static const std::vector< std::string > paths =
+  {
+    REMILL_BUILD_SEMANTICS_DIR_X86,
+    REMILL_BUILD_SEMANTICS_DIR_AARCH32,
+    REMILL_BUILD_SEMANTICS_DIR_AARCH64,
+    REMILL_BUILD_SEMANTICS_DIR_SPARC32,
+    REMILL_BUILD_SEMANTICS_DIR_SPARC64,
+    REMILL_INSTALL_SEMANTICS_DIR,
     "/usr/local/share/remill/" MAJOR_MINOR "/semantics",
     "/usr/share/remill/" MAJOR_MINOR "/semantics",
     "/share/remill/" MAJOR_MINOR "/semantics",
-};
+  };
+  return paths;
+}
 
-}  // namespace
+using maybe_path_t = std::optional< std::filesystem::path >;
 
-// Find the path to the semantics bitcode file associated with `FLAGS_arch`.
-std::string FindTargetSemanticsBitcodeFile(void) {
-  return FindSemanticsBitcodeFile(FLAGS_arch);
+maybe_path_t IsSemanticsBitcodeFile(
+  std::string_view dir,
+  std::string_view arch)
+{
+  auto path = std::filesystem::path(dir) / (std::string(arch) + ".bc");
+  return (std::filesystem::exists(path)) ? std::make_optional(std::move(path)) : std::nullopt;
 }
 
-// Find the path to the semantics bitcode file associated with `REMILL_ARCH`,
-// the architecture on which remill is compiled.
-std::string FindHostSemanticsBitcodeFile(void) {
-  return FindSemanticsBitcodeFile(REMILL_ARCH);
+}  // namespace
+
+std::optional<std::string> _FindSemanticsBitcodeFile(std::string_view arch,
+                                                     const std::vector< std::string > &dirs) {
+ for (const auto &dir : dirs)
+    if (auto sem_path = IsSemanticsBitcodeFile(dir, arch))
+        return sem_path->string();
+  return {};
 }
 
-// Find the path to the semantics bitcode file.
-std::string FindSemanticsBitcodeFile(std::string_view arch) {
-  if (!FLAGS_semantics_search_paths.empty()) {
-    std::stringstream pp;
-    pp << FLAGS_semantics_search_paths;
-    for (std::string sem_dir; std::getline(pp, sem_dir, ':');) {
-      std::stringstream ss;
-      ss << sem_dir << "/" << arch << ".bc";
-      if (auto sem_path = ss.str(); FileExists(sem_path)) {
-        return sem_path;
-      }
-    }
+std::optional<std::string> FindSemanticsBitcodeFile(std::string_view arch,
+                                     const std::vector< std::string > &dirs,
```

Comment:
Take in a vector of `std::filesystsem::path`?

Reply:
To make transition easier I wanted to expose older type of API, but I guess I can change to return `path` as well in one swoop.